### PR TITLE
542_01_Matrix

### LINKED ISSUE #66: 542_01_Matrix
## Summary
542_01_Matrix
## Goal
542_01_Matrix
## Todo

- [ ] add 542_01_Matrix
- [ ] write time and space complexity

## Deadline
2/8
## Parent issue
https://github.com/okuda-seminar/Engineering/issues/599
## References
None

## Notes
None

### PR DESCRIPTION
**Pull request by Rwatana**
Please check my code.

### CODE REVIEW

File: 542_01_Matrix.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Time: O(n**2)
+# Space: O(1)
```
Comment (ryujumoto) [inline 542_01_Matrix.py]: Time complexity and space one are wrong.

File: 542_01_Matrix.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+
+class Solution(object):
+
+    def updateMatrix(self, matrix):
```
Comment (ryujumoto) [inline 542_01_Matrix.py]: Add type annotation.

**Review by ryujumoto — COMMENTED** [review]
Fix complexity and add type annotation.
Also, you solved this problem with dynamic programming, but there is another solution using BFS. Try another solution, and push it.

File: 542_01_Matrix.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#                         matrix[x][y] = min(dp[x][y], 1+min(matrix[x+1][y],
+#                                                            matrix[x][y+1]))
+
+#         return matrix
+
+# BFS
+# n = len(mat)
+# m = len(mat[0])
+# Time: O(n*m)
+# Space: O(1)
```
Comment (ryujumoto) [inline 542_01_Matrix.py:45]: Space complexity is wrong.

**Review by ryujumoto — COMMENTED** [review]
Done.

**Comment by Rwatana** [comment]
I revised and add BFS. Please check.

**Pull request closed without merging**